tahoe_sites/models.py

```diff
@@ -43,6 +43,31 @@ def __str__(self):
             organization=self.organization.short_name,
         )
 
+    @classmethod
+    def is_user_already_mapped(cls, user):
+        """
+        Check if the user already mapped to an organization or not (regardless of active/admin statuses)
+
+        :param user: User to check
+        :return: <True> if the user already mapped to an organization, <False> otherwise
+        """
+        return cls.objects.filter(user=user).count() > 0
+
+    def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
+        """
+        Override save to control user uniqueness rule
+
+        :param force_insert: inherited
+        :param force_update: inherited
+        :param using: inherited
+        :param update_fields: inherited
+        :return: inherited
+        """
+        if not self.pk and self.is_user_already_mapped(user=self.user):
+            raise IntegrityError('Cannot add user to organization. User already added to an organization!')
+
+        super().save(force_insert, force_update, using, update_fields)
```

Comment:
Overriding `save()` may cause un-intended issues -- check the [Django docs for the better alternative](https://docs.djangoproject.com/en/4.0/topics/signals/). Please add it as a `pre_save` signal:

```py


@receiver(pre_save, sender=UserOrganizationMapping)
def prevent_adding_user_to_two_organizations(sender, instance, **kwargs):
        """
        Prevent adding the same user to two organizations.
        """
        if not self.pk and self.is_user_already_mapped(user=self.user):
            raise IntegrityError('Cannot add user to organization. User already added to an organization!')
```

I've added the change in: https://github.com/appsembler/tahoe-sites/pull/25

Reply:
Many Thanks